README.md

```diff
@@ -10,21 +17,68 @@ files) from the respective projects. An overview is listed at
 https://github.com/Sensirion/info#repositories
 
 ## Repository content
-* `common` folder (shared functions and global defines)
-* `shdlc` folder (implementation of the SHDLC communication and UART HAL
-  including sample code for some platforms)
-* `i2c` folder (implementation of the I2C communication)
-* `i2c/hw` folder (stub for implementation of the hardware I2C HAL,
-  including sample code for some platforms)
-* `i2c/sw` folder (stub for implementation of the software I2C HAL
-  (bit-banging GPIOs), including sample code for some platforms)
-* `tests-common-config` folder (configuration files for tests, used by driver repositories)
+
+The source code in this repository is spilt into three parts: Common code,
+I2C protocol and SHDLC protocol.
+
+### Common
+
+In the `common/` folder are three files. `senirion_config.h` is the general
+configuration file. Depending on your hardware you will need to edit this.
+For more information, see the section named `sensirion_config.h` further down
+in this document. There are also `sensirion_common.[ch]`. These contain common
+helper functions used by both UART and I2C sensor drivers.
+
+### I2C
+
+In the `i2c/` folder is the implementation of the protocol used by Sensirion
+sensors for I2C communication. This implementation is spilt into two parts.
+The implementation of the high level protocol in the files `sensirion_i2c.[ch]`
+and a hardware abstraction layer in `senirion_i2c_hal.[ch]`.
+In `sensirion_i2c.[ch]` all the implementation details (CRC calculation, byte order
+swapping, ...) of Sensirion's I2C communication are implemented. The finished frame
+is then sent to the attached hardware using the Hardware Abstraction Layer (HAL) in
+the other two files. Since the way how to send out this frame depends on underlying
+hardware and operating system, we provide different implementations for that part
+of the code. You can find these in the folder `i2c/sample-implementations/` for the
+most common platforms. We even provide a implementation for software I2C using
+bit-banging on GPIOs in `i2c/sample-implementations/GPIO_bit_banging/`. If your
+platform isn't currently there, you can use the provided template structure in the
+`i2c/` folder to implement it yourself. We're very happy to review and include more
+architectures in the form of a pull request on GitHub.
+
+### SHDLC
+
+In the `shdlc` folder is the implementation of the protocol used by Sensirion
+sensors for UART communication. These sensors use Sensirions own SHDLC protocol.
```

Comment:
```suggestion
sensors for UART communication. These sensors use Sensirion's own SHDLC protocol.
```

You use Sensirion's everywhere else. I'd also be fine with using Sensirions everywhere.